packages/mobile/src/home/saga.ts

```diff
@@ -42,6 +42,8 @@ export function* refreshBalances() {
   yield put(fetchDollarBalance())
   yield put(fetchGoldBalance())
   yield put(fetchSentEscrowPayments())
+  // TODO: Find a better home for this
+  yield put(fetchRecentTxRecipients())
```

Comment:
Like you'd TODO says, this doesn't seem like the right trigger for this. Should it happen on feed update?

Reply:
What action does that happen on? I originally tried `NEW_TRANSACTION_IN_FEED` but `addressToE184PhoneNumber` wasn't updated yet so it was happening slightly too early

Reply:
I think `NEW_TRANSACTION_IN_FEED` is the closet we have to when a send or receive is complete. Can you clarify why it didn't work for this?

Reply:
The query that triggers `NEW_TRANSACTION_IN_FEED` has a very tight loop and picks up on a send transaction before the mapping of address to phone number has been stored in `addressToE164Number`. This is a problem because `addressToE164PhoneNumber` is  needed to do the lookup from address -> e164PhoneNumber -> recipient.

This only happens on the first transaction to a particular recipient. If this is a transaction to someone you've already transacted with (i.e., the address to phone number mapping already exists in `addressToE164PhoneNumber`), then this isn't an issue

Reply:
Ah I think the problem is that you need to wait for this saga to run first:
https://github.com/celo-org/celo-monorepo/blob/master/packages/mobile/src/identity/saga.ts#L94

So how about trigger your new saga whenever UPDATE_E164_PHONE_NUMBER_ADDRESSES *or* NEW_TRANSACTION_IN_FEED happens?

Reply:
I think that combo will do the trick. Will test it out now